src/server/pps/server/api_server.go

```diff
@@ -1767,12 +1771,51 @@ func (a *apiServer) pipelineManager(ctx context.Context, pipelineInfo *pps.Pipel
 				}
 			}
 
+			newBranch := branchSet.Branches[branchSet.NewBranch]
```

Comment:
What's happening in this block of code?  Are we trying to find the parent job?  Maybe add some comments.  Also this code looks strikingly similar to the block of code above it, so I wonder if there's some opportunities for refactoring here.